src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -26,11 +26,11 @@ public class AddCommandParser implements Parser<AddCommand> {
      * and returns an AddCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public AddCommand parse(String args, String selectedDeckName) throws ParseException {
+    public AddCommand parse(String args, Deck selectedDeck) throws ParseException {
```

Comment:
Actually, I agree with @ rockman007372, the parse method should only take in userInput and not additional parameters like selectedDeck. Cause a parser's role is to parse user commands.

 I suggest passing the card creation to AddCommand instead of this class.

Reply:
okay agreed

Reply:
agreed!